spec/ysh-stdlib-synch.test.sh

```diff
@@ -0,0 +1,56 @@
+# spec/ysh-stdlib
+
+## our_shell: ysh
+
+#### semaphore
+source --builtin synch.ysh
+
+var s = semaNew(1)
+{ 
+  sleep 0.5
+  call semaDown(s)
+  echo 1
+} &
```

Comment:
Style note: `fork { myblock }` is preferred in YSH  (we could omit the `&` operator)


we also have `forkwait { myblock }`